### PR TITLE
Alternate Jobs Database

### PR DESCRIPTION
**Pull request by west3436**
## What this does

Adds the Alternate Jobs Database for 6 TC (3 if you're the HOP). Installing this into the Labor Administration Console will scramble Central Command's jobs database, resulting in latejoiners being **given the option to** play as one of the following roles instead of the role they chose:

Chiropractor
Dog Walker
Psychologist
Scuba Diver
Plumber
Dentist
Management Consultant
Wedding Planner
Lifeguard
Insurance Salesman
Cable Guy
Wood Identifier
Interior Designer
Sommelier
Bathroom Attendant
Welding Fuel Tank Refiller
Historical Reenactor

Unique sprites courtesy of @BomberBro:
scuba
wftr

## Why it's good

HOP traitor item which can cause some fun RP and a moderate amount of trolling. Kicking the computer will remove the software.

## Changelog

:cl:
 * rscadd: Adds the Alternate Jobs Database Disk which causes new recruits to join as silly roles.

### CODE REVIEW

**Comment by Eneocho** [comment]
Do you plan to make it so they have modified access or are they only renamed and that's it? If the latter, could stand to have a specialized custom one for the "head of security engineering" "head of security clown" "head of security chef" experience.

**Comment by west3436** [comment]
> Do you plan to make it so they have modified access or are they only renamed and that's it? If the latter, could stand to have a specialized custom one for the "head of security engineering" "head of security clown" "head of security chef" experience.

All alt titles have assistant access.

**Comment by skelesexualstephen** [comment]
This would discourage late joining and be bad for server health. I for one would probably just log off if this happened on a round with no hop.

**Comment by Dacendeth** [comment]
Would latejoin implanted people spawn without an implant?

**Comment by west3436** [comment]
> Would latejoin implanted people spawn without an implant?

Yes

**Comment by Dacendeth** [comment]
This also shouldn't work against latejoin traders if it does.

**Comment by Kurfursten** [comment]
I think this is a cute idea (similar to an idea I had of emagging the console to enable some jobs like Cyborgification Volunteer) but forcing latejoiners to have assistant access is absolutely demoralizing. Imagine you join to do Xenoarch. You want to do xenoarch, you're just here to dig. Whoopsie now you have to spend the next 15 minutes trying to find someone willing to fix your ID. You can apply this situation to any jobbie role.

Now imagine someone pulls this out during a high intensity shift like revs, blob, or ops. You think you can get an ID change in revs?

**Comment by west3436** [comment]
> I think this is a cute idea (similar to an idea I had of emagging the console to enable some jobs like Cyborgification Volunteer) but forcing latejoiners to have assistant access is absolutely demoralizing. Imagine you join to do Xenoarch. You want to do xenoarch, you're just here to dig. Whoopsie now you have to spend the next 15 minutes trying to find someone willing to fix your ID. You can apply this situation to any jobbie role.
> 
> Now imagine someone pulls this out during a high intensity shift like revs, blob, or ops. You think you can get an ID change in revs?

The inconvenience is... the point as it's an antagonistic tool. This isn't round removing anyone - in the worst case you have to RP as your new role and in 75% of the cases you'll just go to the HOP desk for an access change.

**Comment by skelesexualstephen** [comment]
> > I think this is a cute idea (similar to an idea I had of emagging the console to enable some jobs like Cyborgification Volunteer) but forcing latejoiners to have assistant access is absolutely demoralizing. Imagine you join to do Xenoarch. You want to do xenoarch, you're just here to dig. Whoopsie now you have to spend the next 15 minutes trying to find someone willing to fix your ID. You can apply this situation to any jobbie role.
> > Now imagine someone pulls this out during a high intensity shift like revs, blob, or ops. You think you can get an ID change in revs?
> 
> The inconvenience is... the point as it's an antagonistic tool. This isn't round removing anyone - in the worst case you have to RP as your new role and in 75% of the cases you'll just go to the HOP desk for an access change.

Does this leave any counter play? Can you track down the person who did it and kill them? Does it leave evidence behind for someone to investigate? Can it be fixed or corrected somehow? It is it just a magic button you get to press to fuck up an entire rounds crew with no risk?

When the server is just warming up for the day you have a round with 5 people that slowly snowballs to 20+ how many of those 15 late joiners will just nope out?

**Comment by Kurfursten** [comment]
>It antagonizes people immediately as soon as they spawn, before they get off the shuttle

You have just mentally reframed this PR to be spawncamping in my mind. I'm going to have to call this a hard no since it's in violation of server rules.

**Comment by Eneocho** [comment]
I'd say enable emagging the console for the funny roles for latejoins then.

**Comment by BomberBro** [comment]
It's not spawncamping even according to the official rules, lame cunts killing themselves because they can't be bothered with any challenges doesn't count as the traitor killing people as soon as they arrive. 
spawncamping

**Comment by jwhitak** [comment]
How do you undo this in game?

**Comment by west3436** [comment]
> How do you undo this in game?

I mentioned it in the OP - just kick the computer. The disk is reusable.

**Comment by Kurfursten** [comment]
I've retracted my opinion on this because it now just gives people the option to join as silly assistant titles. I don't think this is even controversial anymore in its current state, but I could be wrong. It does function as an instant +3 assistant slots but that's hardly as chaotic as the original proposal.

File: code/game/jobs/job_controller.dm [hunk 1]
```diff
@@ -581,3 +600,12 @@ var/global/datum/controller/occupations/job_master
 
 		tmp_str += "HIGH=[level1]|MEDIUM=[level2]|LOW=[level3]|NEVER=[level4]|BANNED=[level5]|YOUNG=[level6]|-"
 		feedback_add_details("job_preferences",tmp_str)
+
+/datum/controller/occupations/proc/altjobprompt(var/newrank,var/oldrank,var/mob/user)
+	var/turf/oldloc = get_turf(user)
+	user.forceMove(null)
+	if(alert(user,"Central Command had a mix-up and is attempting to send you to the station as \an [newrank]! Would you like to correct them?",,"Yes - play as [oldrank]","No - play as [newrank]") == "Yes")
```
Comment (Kurfursten) [inline code/game/jobs/job_controller.dm]: This will always return false because Yes does not match either option

**Comment by angryturnip** [comment]
Though I'm not sure it'd be any more successful, it would also be neat to see jobs combined via emagging the jobs console, giving access to both the jobs. I.e. Sciengineer, Chief Mime Officer, ect.

**Comment by Eneocho** [comment]
Head of Clown
Clownist
Medical Clown
Research Clown
Chief Clown Officer
Clowntain
Clownssistant
Clown Technician
Atmospherics Clown
Clown Officer
Clown of Personnel

**Comment by SonixApache** [comment]
optional makes this 👌

**Comment by west3436** [comment]
> optional makes this 👌

babby mode

**Comment by Inorien** [comment]
seems kinda weird to give people an option /at all/

>hmm today i will engineer
>join as Engineer
>Sleve McDichael, Fursuit Cleaner, has arrived on the station

giving people the option to join as a toast inspector or whatever rather than what they started Byond wanting to play would just discourage them from joining, I don't think there'll be many people immediately going braindead in arrivals because they got switcheroo'd by the spicy labour admin console

**Comment by west3436** [comment]
> seems kinda weird to give people an option /at all/
> 
> >hmm today i will engineer
> >join as Engineer
> >Sleve McDichael, Fursuit Cleaner, has arrived on the station
> 
> giving people the option to join as a toast inspector or whatever rather than what they started Byond wanting to play would just discourage them from joining, I don't think there'll be many people immediately going braindead in arrivals because they got switcheroo'd by the spicy labour admin console
> 

Take that up with @Kurfursten as he deemed that to be the same as killing them on the arrivals shuttle, so I won't be doing it.

**Comment by Inorien** [comment]
> Take that up with @Kurfursten as he deemed that to be the same as killing them on the arrivals shuttle, so I won't be doing it.

i think the issue there is that most of the time when you get antagonised otherwise its because your mario was in the wrong place at the wrong time, but presenting people an option to play as a glue cleaner **instead of** a proper role is where the antagonising spills out of the game, which i think is bad
my suggestion is to present the normal list but not apply the role switch until after the player has chosen a normal role, so the fake job only appears after loading into the shuttle

**Comment by Kurfursten** [comment]
I'm satisfied that this is opt-in enough to not be shuttle camping. Inorein's approach might be better overall for getting someone who WANTS to be a Toast Inspector but this implementation is fine right now.

**Comment by BomberBro** [comment]
\>I don't think there'll be many people immediately going braindead in arrivals because they got switcheroo'd by the spicy labour admin console
Frankly this says more about players refusing to roleplay than the PR itself. It's NOT spawncamping to be given a different, not crucial job with your arrival. If you want to get a job change, get a job change. The pearl-clutching over this PR is absolutely inane, traitors can ruin rounds in far worse ways than this and them dumping their TC to effectively just reduce someone's access (after they get important console access) is hardly gonna be a major blow to the station.

**Comment by Eneocho** [comment]
The same way suiciding doesn't count as human harm to AI, a player ghosting/suiciding because he got the funny role instead of his intended one shouldn't count as spawncamping.
It's a really weird implementation now, you pay TC to give latejoins the OPTION to be antagonized? What? And just like inorien said, having the antagging spill out into the player, instead of strictly in-game is really fucking weird.

**Comment by jwhitak** [comment]
Trying to join as one role and getting another will demoralize players who just spawned in and possibly damage pop. While, yeah, skill issue git gud etc? I don't think it's a good change for the game. Just let people join as who they want to play as. Also, how do you plan to log these jobs joining? If they cannot select their own role, then there's no point in using standard job logging for them.

If this PR does get merged, at least give them the heads up on the JOIN ROUND button that they're going to get a random job, so that they can at least try and play into it. Give 'em some agency.

Edit: Also, how will this affect specific roles such as Clowns who start off with the clumsy gene? Will they just not get the gene?

**Comment by Kurfursten** [comment]
> The same way suiciding doesn't count as human harm to AI, a player ghosting/suiciding because he got the funny role instead of his intended one shouldn't count as spawncamping. It's a really weird implementation now, you pay TC to give latejoins the OPTION to be antagonized? What? And just like inorien said, having the antagging spill out into the player, instead of strictly in-game is really fucking weird.

It's pay 6TC (3 TC HoP) to add 3 assistants, which is a reasonable amount of chaos for 3TC. Compare to 2TC chamelon jumpsuit, 4TC DNA scrambler, 4 TC Camera Bugs.

**Comment by Eneocho** [comment]
>It's pay 6TC (3 TC HoP) to add 3 assistants, which is a reasonable amount of chaos for 3TC. Compare to 2TC chamelon jumpsuit, 4TC DNA scrambler, 4 TC Camera Bugs.

The difference is that the jumpsuit, scrambler, camera bugs all have a use. You can disguise/hide with the jumpsuit, scramble your genes for disguising further, put the camera bugs on important players for spy stuff (find the perfect time to telesci a nitro and instantly detonate it, for example).
Those 3 assistants are just regular crew. You're literally paying TC to reward the crew with extra greyshirt slots, which makes zero sense to me. Sure the greyshirts could cause chaos, but they could've caused chaos in any of the roles they would've gotten instead of the greyshirt alt role.
I'd support this implementation if it was an emag act on the job machine (yeah yeah emag creep whatever), since that does not reward the crew in exchange for TC, but rather gives an extra, less harmful, use to an already existing antag tool.

**Comment by BomberBro** [comment]
Two bad comments I want to reply to on this:
\> Can it be fixed or corrected somehow?
Yes.
\>? Does it leave evidence behind for someone to investigate? 
Yes, use a DNA scanner on the HoP's doors or the console you fucking moron. 
\>Does this leave any counter play? Can you track down the person who did it and kill them?
Wait until you find out about the subspace tunneler or simply shooting a revolver down a hallway. The fact that your immediate reaction to a mild antagonism like that is to think of murder is really showing how much these people who allegedly care so much about their roles wish to roleplay. 

\>This will break job logging!!!
This is such a non issue it's almost nonsensical to use it as a reason to dislike this PR. Considering in its current state it will see rare use if any since it doesn't really even antagonize, the amount of times the fake jobs will get into your UNOFFICIAL job stats will be minimal, and for admin purposes it simply will not matter. 
\>Just let people join as who they want to play as.
The current iteration of this PR which was edited to say this in bold letters before you posted already says that currently it's **optional**. I'm not sure why you downvoted and said this despite that. 

Again though, I'm astounded by the audacity of the people here, alleged roleplayers, who are saying "uh yeah i'd rather just not play or kill myself if I tried to join in as a doctor and it made me a scuba diver". This is not bay, there is no skill system, you can still do doctoral activities just as good as anyone else so roll with the fucking improv you babies. The ONLY real problem I can see is if command try to join in and I wouldn't mind making it forced for everyone EXCEPT command roles.

**Comment by Kurfursten** [comment]
> The current iteration of this PR which was edited to say this in bold letters before you posted already says that currently it's **optional**. I'm not sure why you downvoted and said this despite that.

Bomber chastising someone for missing changed bolded text while himself not noticing that it **hasn't** changed in the official changelog. Embarrassing...

**Comment by BomberBro** [comment]
@Kurfursten If you're suggesting that the changelog is causing genuine confusion amongst the githubbers reading the PR then perhaps you, the collab in charge of quality control, should have told west to change it when you allowed him to reopen it with your _satisfaction_. At any rate I'm not the one who downvoted the PR apparently without reading the post or the comments despite posting in them multiple times and I'm also not here for a hot-take war that I could get off of /v/. 

The only valid point imo I've seen to detract from this PR is that it might do weird things with latejoins like clowns that get special genes and my personal point that it shouldn't affect command roles since it's MUCH harder to convince someone to fix that. The rest of the points against this PR are completely nonsensical "what if" scenarios or bad faith arguments.

\>Imagine you join to do Xenoarch. You want to do xenoarch, you're just here to dig. Whoopsie now you have to spend the next 15 minutes trying to find someone willing to fix your ID. (...) Now imagine someone pulls this out during a high intensity shift like revs, blob, or ops. You think you can get an ID change in revs?
Are you saying that this PR is going to.... hinder your ability to do your job during nukeops or revolution? As if the round type won't? In these scenarios you won't have 15 minutes **at all** before the round ends, and you're not gonna get much done as those jobs even with your access.  Especially nukeops. You're bringing up scenarios of the station in chaos as reasons why you wouldn't be able to get your access fixed, but fail to mention the round is likely going to be over in the next 5-10 minutes anyway. There's just not really a scenario where it matters. And on blob typically all access is thrown around for free so that shouldn't even be an issue.

**Comment by Kurfursten** [comment]
> without reading the post 

But you didn't read the post. You stopped at the first line and never made it to the changelog.

> Perhaps you shouldn't have let him reopen the PR until you proofread his post for him

This is what passes for BomberHarris "not here for a hot take war".

**Comment by BomberBro** [comment]
\>But you didn't read the post. You stopped at the first line and never made it to the changelog.
I mentioned that west posted the changes in the comments and his post so I clearly did read it. When I personally upvoted it I read the whole thing (I still would upvote the original implementation, though perhaps with a caveat) and cast my vote. Dilt clearly did not. That's why I mentioned it and you'd have to be an olympic tier mental gymnast to say that someone would reasonably think the PR was still like that because they read only the changelog blurb and not the comments or the main body of the post, especially if they have their own posts in the comment section. The main reason I even mentioned it was to show people were downvoting it without looking too closely, which it proves. 

\>This is what passes for BomberHarris "not here for a hot take war".
I gave you a meaningful, detailed reply to your comment with your concern immediately after and you completely ignored it in favor of making another hot take. Your replies are more indicative of a 4chan poster even when I'm the one using chevrons.

**Comment by Eneocho** [comment]
this fucking PR, man

**Comment by Pastamadness** [comment]
Sounds lame and pointless. This gimmick is so lame that it should be available to any HoP instead of being locked behind traitor without antagonizing anyone. Rainbow alerts are more interesting and disrupting to the round and any HoP can enact those.
Add Potash Maker, by the way.

**Comment by west3436** [comment]
@BomberBro is working on sprites and maybe outfit datums for this while I am tied up with fire rework; this is not derelict YET

File: code/game/jobs/job_controller.dm (hunk 1 above)
Comment (west3436) [reply code/game/jobs/job_controller.dm]: fixed in a recent commit

File: code/modules/clothing/suits/miscellaneous.dm [hunk 2]
```diff
@@ -1164,3 +1164,18 @@ var/list/tag_suits_list = list()
...
+	name = "scuba suit"
+	desc = "A watertight swimming suit used to support a Self-Contained Underwater Breathing Apparatus."
+	icon_state = "scuba"
+	item_state = "scuba"
+	origin_tech = Tc_MATERIALS + "=2;" + Tc_ENGINEERING + "=1"
+	w_class = W_CLASS_LARGE//bulky item
+	gas_transfer_coefficient = 0.90
+	permeability_coefficient = 0.50
+	body_parts_covered = ARMS|LEGS|FULL_TORSO|FEET|HANDS
+	allowed = list(/obj/item/weapon/tank/emergency_oxygen,/obj/item/weapon/tank/emergency_nitrogen)
```
Comment (Eneocho) [inline code/modules/clothing/suits/miscellaneous.dm]: the plasmeme can also have the scuba suit, shouldn't this also have the emergency plasma tank as an allowed tank?

Comment (west3436) [reply code/modules/clothing/suits/miscellaneous.dm]: Good catch thanks

File: code/game/machinery/computer/labor.dm [hunk 3]
```diff
@@ -143,9 +143,29 @@ var/list/labor_console_categories = list(
...
 			verified(user)
+	if(istype(W,/obj/item/weapon/disk/jobdisk))
+		to_chat(user, "<span class='notice'>You begin installing the alternate database.</span>")
+		if(!user.drop_item(W))
+			to_chat(user, "<span class='warning'>You can't let go of \the [W].</span>")
+			return
+		W.forceMove(src)
+		if(do_after(user,src,30))
+			playsound(src, 'sound/machines/ping.ogg', 35, 0, -2)
+			to_chat(user, "<span class='notice'>Alternate jobs  database successfully installed.</span>")
```
Comment (DrSnips) [inline code/game/machinery/computer/labor.dm]: Extra space between jobs and database

File: code/datums/outfit/special_outfits/alternate.dm [hunk 4]
```diff
@@ -0,0 +1,840 @@
...
+	pda_slot = slot_l_store
+	id_type = /obj/item/weapon/card/id/engineering
+
+/datum/outfit/cableguy/post_equip(var/mob/living/carbon/human/H)
+	..()
+	H.put_in_hands(new /obj/item/stack/cable_coil)
+
+/datum/outfit/woodidentifier
+
+	outfit_name = "Research Director"
```
Comment (DrSnips) [inline code/datums/outfit/special_outfits/alternate.dm]: Should this be RD or wood IDer?

Comment (west3436) [reply code/datums/outfit/special_outfits/alternate.dm]: Good catch - thanks!

**Review by jwhitak — CHANGES_REQUESTED** [review]
For the most part, this looks great, and my issue regarding being forced into a role you didn't want is resolved with the popup.

Some of the options are kinda questionable (look at how big that welder tank is! or how much money the insurance agent gets!), but it's all in good fun.

The only change I can think of would be to the very thing that appeased my view of the PR. How difficult would it be to add an option to cancel out of joining when offered the job? Some players might want to go back to character creation and come up with a gimmick. If that's too difficult to add, just mark this review resolved and we're good to go.

**Review by jwhitak — APPROVED** [review]

**Comment by jwhitak** [comment]
Resolved on Discord(tm). Future PR material.

**Comment by west3436** [comment]
> For the most part, this looks great, and my issue regarding being forced into a role you didn't want is resolved with the popup.
> 
> Some of the options are kinda questionable (look at how big that welder tank is! or how much money the insurance agent gets!), but it's all in good fun.
> 
> The only change I can think of would be to the very thing that appeased my view of the PR. How difficult would it be to add an option to cancel out of joining when offered the job? Some players might want to go back to character creation and come up with a gimmick. If that's too difficult to add, just mark this review resolved and we're good to go.

If they didn't want to RP as the alt title I can't imagine they'd want to RP as a foil to the alt title (and even still, the next person to join wouldn't be presented with the same alt title that they were).

**Pull request merged**